portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,42 +13,87 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
+import com.google.sps.data.Task;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
+import java.lang.Integer;
+import com.google.gson.Gson;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comment data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   
-  private List<String> facts;
+  private ArrayList<String> comments;
+  private static final Gson GSON = new Gson();
 
   @Override
   public void init() {
-    facts = new ArrayList<>();
-    facts.add(
-        "My favorite color is green. "
-            + "I also like light blue.");
-    facts.add("My parents made up my name based on my Chinese one.");
-    facts.add("My birthday is on Halloween.");
-    facts.add("I have the best younger brother!");
-    facts.add("My favorite food is pasta.");
-    facts.add("I'm a cat person. I like dogs, but I like cats more.");
-    facts.add(
-        "I didn't have any experience with "
-            + "computer science in high school.");
-    facts.add("I play the acoustic guitar.");
+    comments = new ArrayList<>();
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String fact = facts.get((int) (Math.random() * facts.size()));
-    response.setContentType("text/html;");
-    response.getWriter().println(fact);
+    Query query = new Query("Task");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    // datastoreService.prepare(query).asList(FetchOptions.Builder.withLimit(10));
+
+    // Loop over entities.
+    List<Task> tasks = new ArrayList<>();
+    int max = Integer.parseInt(request.getParameter("max"));
+    int i = 0;
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+      tasks.add(new Task(name, text, timestamp));
+      i += 1;
+      if (i == max) break;
```

Comment:
you can actually get rid of the i variable and just do tasks.size() instead

another nit: for if/else statements, always add the { } even if it's optional (I think the reasoning was it's very easy to accidentally introduce a bug if you decide to add more statements to the if and forget to add braces). the style guide literally only has one line about this so it's easily missed: https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used